src/vue/common/CollectionLoader.spec.js

```diff
@@ -0,0 +1,209 @@
+import CollectionLoader from './CollectionLoader'
+import { shallowMount, createLocalVue } from '@vue/test-utils'
+import { ErrorHandler } from '@/js/helpers/error-handler'
+import { Helper } from '@/js/helpers/test-helper'
+
+const localVue = createLocalVue()
+localVue.filter('globalize', sinon.stub())
+
+describe('CollectionLoader component test', () => {
+  const firstPageLoader = sinon.stub().resolves({
+    data: [],
+    fetchNext: () => {}
+  })
+  afterEach(() => {
+    sinon.restore()
+  })
+
+  it('loadPage calls provided loaderFn', async () => {
+    const wrapper = shallowMount(CollectionLoader, {
+      propsData: {
+        firstPageLoader
+      },
+      localVue
+    })
+    const spy = sinon.stub().resolves({
+      data: [],
+      fetchNext: () => {}
+    })
+
+    await wrapper.vm.loadPage('event', spy)
+
+    expect(spy.calledOnce).to.be.true
+  })
+
+  it('loadPage properly emits event with correct payload', async () => {
+    const wrapper = shallowMount(CollectionLoader, {
+      propsData: {
+        firstPageLoader
+      },
+      localVue
+    })
+    const data = []
+    const event = 'event'
+    await wrapper.vm.loadPage(event, sinon.stub().resolves({
+      data
+    }))
+    expect(Helper.isEmittedOnce(wrapper, event, data)).to.be.true
+  })
+
+  it('loadPage saves fetchNext returned in response', async () => {
+    const fetchNext = () => {}
+    const wrapper = shallowMount(CollectionLoader, {
+      propsData: {
+        firstPageLoader
+      },
+      localVue
+    })
+
+    const spy = sinon.stub().resolves({
+      fetchNext,
+      data: []
+    })
+
+    await wrapper.vm.loadPage('event', spy)
+
+    expect(wrapper.vm.nextPageLoader).to.equal(fetchNext)
+  })
+
+  it('isCollectionFetched false when data is not fully loaded', async () => {
+    const fetchNext = () => {}
+    const data = []
+    const pageLimit = 10
+    const wrapper = shallowMount(CollectionLoader, {
+      propsData: {
+        firstPageLoader: sinon.stub().resolves({
+          data,
+          fetchNext
+        }),
+        pageLimit
+      },
+      localVue
+    })
+    data.length = pageLimit + 1
+    const spy = sinon.stub().resolves({
+      fetchNext,
+      data
+    })
+
+    await wrapper.vm.loadPage('event', spy)
+
+    expect(wrapper.vm.isCollectionFetched).to.be.false
+  })
+
+  it('isCollectionFetched true when data is fully loaded', async () => {
```

Comment:
`loadPage sets isCollectionFetched to true ...`

Reply:
ok